### PR TITLE
implementedBookWeb

### PR DESCRIPTION
**Pull request by shumYevh**

### CODE REVIEW

File: src/main/java/org/example/bookweb/models/Book.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String title;
+    private String author;
+    @Column(unique = true)
+    private Long isbn;
+    private Double price;
+    private String description;
```
Comment (okuzan) [inline src/main/java/org/example/bookweb/models/Book.java]: enforce nullability constraints

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]
I think you misunderstood the previous mentor's comment, he asked you to add a nullable constraint to the required fields

**Review by olekskov — APPROVED** [review]

File: src/main/java/org/example/bookweb/service/BookServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+@RequiredArgsConstructor
+@Service
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto add(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        book.setIsbn(new Random().nextLong(1000000000));
```
Comment (olekskov) [inline src/main/java/org/example/bookweb/service/BookServiceImpl.java]: Isbn is assigned during the publishing process. Do not use random, accept this as part of DTO

**Review by olekskov — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/bookweb/service/BookServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        Book book = bookMapper.toModel(requestDto);
+        Book savedBook = bookRepository.save(book);
+        return bookMapper.toDto(savedBook);
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
```
Comment (MAMentorFX) [inline src/main/java/org/example/bookweb/service/BookServiceImpl.java]: Mapstruct supports mapping collections from the box

File: src/main/java/org/example/bookweb/service/BookServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto dto) {
+        Book newBook = bookMapper.toModel(dto);
+        newBook.setId(id);
+        Book existBook = bookRepository.findBookById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't, update book by id: " + id)
+        );
+        newBook.setIsbn(existBook.getIsbn());
+        Book savedBook = bookRepository.save(newBook);
+        return bookMapper.toDto(savedBook);
```
Comment (MAMentorFX) [inline src/main/java/org/example/bookweb/service/BookServiceImpl.java:54]: @Transactional. You make more than one call to DB

**Review by MAMentorFX — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/bookweb/service/BookServiceImpl.java (hunk 3 above)
Comment (shumYevh) [reply src/main/java/org/example/bookweb/service/BookServiceImpl.java]: What should i do here?

**Review by olekskov — APPROVED** [review]

**Pull request merged**